### PR TITLE
DEVOPS-252 - Helm config for auto-APM in all deployments

### PR DESCRIPTION
**Pull request by skmannion**

### CODE REVIEW

File: ndustrial/deployment/templates/deployment.yaml [hunk 1]
```diff
@@ -98,6 +98,10 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
+            - name: DATADOG_TRACE_AGENT_HOSTNAME
```
Comment (kalinon) [inline ndustrial/deployment/templates/deployment.yaml:101]: need to add the other envs as well.

Also should add a values entry for enabling datadog or not. aka:

```
datadog:
  apm:
    enabled: false
    # add more default/customizable apm fields here if needed
```

then wrap the datadog envs in a if block:
```
{{- if .Values.datadog.apm.enabled }}
# put envs here
{{- end }}
```

**Review by kalinon — CHANGES_REQUESTED** [review]

File: ndustrial/deployment/templates/deployment.yaml [hunk 2]
```diff
@@ -98,10 +98,35 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
+            {{- if .Values.datadog.apm.enabled }}
+            {{- if .Values.datadog.apm.trace_agent_hostname }}
```
Comment (kalinon) [inline ndustrial/deployment/templates/deployment.yaml:102]: is this specific to apm?

File: ndustrial/deployment/values.yaml [hunk 3]
```diff
@@ -533,3 +533,20 @@ metrics:
...
+  apm:
+    enabled: false
+    node_env: null
+    agent_host: null
+    env: null
+    version: null
+    profiling_enabled: false
+    logs_injection: null
+    trace_sample_rate:
+    trace_agent_host: null
```
Comment (kalinon) [inline ndustrial/deployment/values.yaml:552]: Do we want these all as `null`? also, are these all apm specific? or can they be used without `apm`? guess im asking if they should be nested under apm.

File: ndustrial/deployment/templates/deployment.yaml [hunk 4]
```diff
@@ -98,10 +98,35 @@ spec:
...
               valueFrom:
                 fieldRef:
                   fieldPath: status.hostIP
+            {{- end }}
+            - name: DATADOG_TRACE_AGENT_HOSTNAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.hostIP
+            - name: NODE_ENV
+              value: {{ .Values.datadog.node_env }}
```
Comment (kalinon) [inline ndustrial/deployment/templates/deployment.yaml:116]: think your pathing is wrong. should be like `.Values.datadog.apm.node_env`

File: ndustrial/deployment/values.yaml (hunk 3 above)
Comment (kalinon) [reply ndustrial/deployment/values.yaml:552]: Also need to add documentation annotations. see: https://github.com/bitnami-labs/readme-generator-for-helm and other vars for examples

**Review by kalinon — CHANGES_REQUESTED** [review]

File: ndustrial/deployment/templates/deployment.yaml (hunk 2 above)
Comment (skmannion) [reply ndustrial/deployment/templates/deployment.yaml:102]: Yes believe all of these configs are. They're all used by the ddtrace library and I think traces are an APM thing

**Review by kalinon — APPROVED** [review]

File: ndustrial/deployment/templates/deployment.yaml (hunk 4 above)
Comment (skmannion) [reply ndustrial/deployment/templates/deployment.yaml]: Yup, knew I needed that change. Caught it in most recent commit.

**Review by kalinon — CHANGES_REQUESTED** [review]
default values need to be set or each env wrapped in an `if` statement

**Pull request closed without merging**